File: create_excel_document.py
r"""
Zero-install Excel spreadsheet creation utility (Python)
Run directly with:
    python create_excel_document.py
    OR
    .\bundle\python\python.exe create_excel_document.py
r"""
import sys
from pathlib import Path

# Add vendor directory to path if not installed globally
vendor_path = Path(__file__).resolve().parent / "vendor"
if vendor_path.exists():
    sys.path.insert(0, str(vendor_path))

import xlsxwriter

def create_excel_report(output_file="spreadsheet_output.xlsx", sheet_name="Metrics"):
    wb = xlsxwriter.Workbook(output_file)
    ws = wb.add_worksheet(sheet_name)

    # Formats
    title_fmt = wb.add_format({'bold': True, 'font_size': 14, 'font_color': '#1F4E79'})
    header_fmt = wb.add_format({'bold': True, 'font_color': 'white', 'bg_color': '#1F4E79', 'border': 1, 'align': 'center'})
    cell_fmt = wb.add_format({'border': 1})
    currency_fmt = wb.add_format({'border': 1, 'num_format': '$#,##0.00'})

    # Write Title
    ws.write('A1', 'Automated Business Analytics Report', title_fmt)

    # Write Headers
    headers = ["Task ID", "Operation", "Department", "Quarterly Budget", "Status"]
    for col_idx, text in enumerate(headers):
        ws.write(2, col_idx, text, header_fmt)

    # Write Data
    data = [
        [201, "Pipeline Sanity Check", "Data Engineering", 4500.00, "Passed"],
        [202, "Automated Document Generation", "AI Operations", 3200.00, "Active"],
        [203, "AlloyDB Health Monitoring", "Infrastructure", 7800.00, "Operational"],
        [204, "Offline Zero-Install Packaging", "Release Management", 2100.00, "Complete"]
    ]

    for row_idx, row in enumerate(data, start=3):
        ws.write(row_idx, 0, row[0], cell_fmt)
        ws.write(row_idx, 1, row[1], cell_fmt)
        ws.write(row_idx, 2, row[2], cell_fmt)
        ws.write(row_idx, 3, row[3], currency_fmt)
        ws.write(row_idx, 4, row[4], cell_fmt)

    ws.set_column('A:A', 12)
    ws.set_column('B:C', 30)
    ws.set_column('D:D', 20)
    ws.set_column('E:E', 15)

    wb.close()
    print(f"Successfully generated Excel spreadsheet: {output_file}")
    return output_file

if __name__ == "__main__":
    create_excel_report("sample_python_spreadsheet.xlsx")

File: create_excel_document.js
/**
 * Zero-install Excel document creation utility (Node.js)
 * Run directly with: node create_excel_document.js
 */
import ExcelJS from 'exceljs';

export async function generateExcelDocument(outputPath = 'output_spreadsheet.xlsx', sheetName = 'Summary Data', data = []) {
    const workbook = new ExcelJS.Workbook();
    workbook.creator = 'AI Autonomous Bundle';
    workbook.created = new Date();

    const sheet = workbook.addWorksheet(sheetName);

    sheet.columns = [
        { header: 'ID', key: 'id', width: 10 },
        { header: 'Task Name', key: 'task', width: 35 },
        { header: 'Category', key: 'category', width: 20 },
        { header: 'Status', key: 'status', width: 18 },
        { header: 'Execution Cost', key: 'cost', width: 18 },
    ];

    // Style header row
    const headerRow = sheet.getRow(1);
    headerRow.font = { name: 'Calibri', size: 11, bold: true, color: { argb: 'FFFFFFFF' } };
    headerRow.fill = {
        type: 'pattern',
        pattern: 'solid',
        fgColor: { argb: 'FF1F4E79' }
    };
    headerRow.alignment = { vertical: 'middle', horizontal: 'center' };

    const sampleRows = data.length > 0 ? data : [
        { id: 101, task: 'Word Document Generation (.docx)', category: 'Reporting', status: 'Completed', cost: 150.00 },
        { id: 102, task: 'Excel Spreadsheet Automation (.xlsx)', category: 'Analytics', status: 'Completed', cost: 220.50 },
        { id: 103, task: 'AlloyDB Cloud SQL Sanity Checks', category: 'Database', status: 'Verified', cost: 310.00 },
        { id: 104, task: 'Zero-Install Offline Packaging', category: 'Deployment', status: 'Ready', cost: 180.00 },
    ];

    sampleRows.forEach(row => {
        const added = sheet.addRow(row);
        added.getCell(5).numFmt = '$#,##0.00';
    });

    await workbook.xlsx.writeFile(outputPath);
    console.log(`Successfully created Excel spreadsheet: ${outputPath}`);
    return outputPath;
}

if (process.argv[1] && process.argv[1].endsWith('create_excel_document.js')) {
    generateExcelDocument('sample_generated.xlsx');
}
